compiler/testData/codegen/box/collections/inSetWithSmartCast.kt

```diff
@@ -0,0 +1,15 @@
+// WITH_RUNTIME
+
+fun contains(set: Set<Any>, x: Int): Boolean = when {
+    set.size == 0 -> false
+    else -> x in set as Set<Int>
+}
+
+fun box(): String {
+    val set = setOf(1)
+    if (contains(set, 1)) {
+        return "OK"
+    }
+
+    return "Fail"
+}
```

Comment:
This test in not related to Range#contains, but at some point I've got implementation which passed all tests (without this one), but ant build was failed.